controllers/subscription.go

```diff
@@ -22,44 +22,45 @@ import (
 	"log"
 	"reflect"
 
+	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	olmclient "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
-	olmclient "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
-
-	api "github.com/hybrid-cloud-patterns/patterns-operator/api/v1alpha1"
-	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"k8s.io/client-go/kubernetes"
 )
 
 const (
-	subscriptionNamespace = "openshift-operators"
-	applicationNamespace  = "openshift-gitops"
+	subscriptionNamespace  = "openshift-operators"
+	applicationNamespace   = "openshift-gitops"
+	gitopsSubscriptionName = "openshift-gitops-operator"
+	customConfigFile       = "/gitops-config/config.json"
 )
 
-func newSubscription(p api.Pattern) *operatorv1alpha1.Subscription {
-	//  apiVersion: operators.coreos.com/v1alpha1
-	//  kind: Subscription
-	//  metadata:
-	//    name: openshift-gitops-operator
-	//    namespace: openshift-operators
-	//  spec:
-	//    channel: v1.8.x
-	//    installPlanApproval: Automatic
-	//    name: openshift-gitops-operator
-	//    source: redhat-operators
-	//    sourceNamespace: openshift-marketplace
-	//    startingCSV: openshift-gitops-operator.v1.4.1
-	//    config:
-	//      env:
-	//        - name: ARGOCD_CLUSTER_CONFIG_NAMESPACES
-	//          value: "*"
-
-	spec := &operatorv1alpha1.SubscriptionSpec{
-		CatalogSource:          p.Spec.GitOpsConfig.OperatorSource,
-		CatalogSourceNamespace: "openshift-marketplace",
-		Channel:                p.Spec.GitOpsConfig.OperatorChannel,
-		Package:                "openshift-gitops-operator",
-		InstallPlanApproval:    operatorv1alpha1.ApprovalAutomatic,
+func newSubscriptionFromConfigMap(r kubernetes.Interface) (operatorv1alpha1.Subscription, error) {
+	var newSubscription *operatorv1alpha1.Subscription
+
+	// Check if the config map exists and read the config map values
+	cm, _ := r.CoreV1().ConfigMaps(OperatorNamespace).Get(context.Background(), OperatorConfigFile, metav1.GetOptions{})
+
+	if cm != nil {
+		PatternsOperatorConfig = cm.Data
+	}
+
+	var installPlanApproval operatorv1alpha1.Approval
+
+	if configValueWithDefault(PatternsOperatorConfig, "gitops.installApprovalPlan", GitOpsDefaultApprovalPlan) == "Manual" {
+		installPlanApproval = operatorv1alpha1.ApprovalManual
+	} else {
+		installPlanApproval = operatorv1alpha1.ApprovalAutomatic
+	}
+
+	newSpec := operatorv1alpha1.SubscriptionSpec{
```

Comment:
This becomes simpler if you follow the approach of a separate type, I think

Reply:
Done